### PR TITLE
Fixes bots and improves mob explore objective (attemp 2)

### LINKED ISSUE #6600: somes bugs (floorbot )
### Environment
- **Game Version:** 21052107
- **OS Version:** Lubuntu 18.04x64 
- **Linux version** 4.15.0-142-generic (buildd@lgw01-amd64-036) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021

- map : UnRealStation

### Detailed description of bug

Airlockdoor don't open/interact
(only the default ones on the card. the ones created work, on right on the gif)

Peek_22-05-2021_04-10

floorbots, laying floor correctly. but do not lay floor (plating) on grids just floortiles without structure.
and can put floor on catwalk (grill)

Peek_22-05-2021_04-15

Peek_22-05-2021_04-16

we can't climb in a glass table

### LINKED ISSUE #8055: Bots no longer work
-Cleanbots, floorbots and medbots are broken.

-The mob brains were removed in #8025

They no longer start to explore and find targets.

The issue is that they are different to the other mobs as they don't have the MobAi component (as they don't have health, integrity instead)

### PR DESCRIPTION
**Pull request by ilinm1**
screwed up project version in #8589 so making it again

### Purpose
Mobs now actually scan the enviroment searching for their targets. Before they just randomly wandered around hoping to stumble upon something.

Also adds simple animal components to bots, so they are working again.

### Notes:
Closes #6600
Closes #8055
draft

### Changelog:
CL: [Improvement] - Improves mobs explore objective
CL: [Fix] - Fixes bots

### CODE REVIEW

**Comment by corp-0** [comment]
why is it still a draft?

**Comment by ilinm1** [comment]
cause some objectives still no worky, gonna fix it soon hopefully

**Comment by ilinm1** [comment]
I think its ready

File: UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs [hunk 1]
```diff
@@ -79,6 +125,80 @@ public void OnSpawnServer(SpawnInfo info)
...
+			visionCirclePerimeter.Add(new Vector2Int(xc + x, yc - y));
+			visionCirclePerimeter.Add(new Vector2Int(xc - x, yc - y));
+			visionCirclePerimeter.Add(new Vector2Int(xc + y, yc + x));
+			visionCirclePerimeter.Add(new Vector2Int(xc - y, yc + x));
+			visionCirclePerimeter.Add(new Vector2Int(xc + y, yc - x));
+			visionCirclePerimeter.Add(new Vector2Int(xc - y, yc - x));
+		}
+
+		//https://rosettacode.org/wiki/Bitmap/Bresenham%27s_line_algorithm#C.23
+		private void FillCircle(int xc, int yc, int z)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs:173]: Same comment as in your [first PR](https://github.com/unitystation/unitystation/pull/8589/files#r862340599).

`Physics2D.OverlapCircleAll()` no good? Is it slower?

e.g. 

```
var mobsFound = Physics2D.OverlapCircleAll(worldPos, radius, LayerMask.GetMask("Players", "NPC"));
```

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs:173]: Sorry, I see, this is not just for medibots 😛

File: UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs [hunk 2]
```diff
@@ -204,6 +368,23 @@ protected virtual void PerformTargetAction(Vector3Int checkPos)
...
+								if (!IsEmagged && player.playerHealth.OverallHealth < 75)
+								{
+									player.playerHealth.CirculatorySystem.BloodPool.Add(new ReagentMix(HEALING_REAGENT, 5, 283.15f));
+									Chat.AddCommMsgByMachineToChat(gameObject, "Feel better soon.", ChatChannel.Local, Loudness.NORMAL, broadcasterName: "Medibot: ");
+								}
+								else if (IsEmagged) player.playerHealth.CirculatorySystem.BloodPool.Add(new ReagentMix(HARMFUL_REAGENT, 5, 283.15f));
+								else continue;
+							}
+						}
+					}
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs:387]: I see you're just following the existing pattern, but this logic and the rest of the code specific to one mob should really be delegated to some other component and not reside in the controller. But the refactor can be for another time : )

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**